semgrep/semgrep/output.py

```diff
@@ -343,9 +344,14 @@ def close(self) -> None:
             # using this to return a specific error code
             final_error = SemgrepError("", code=FINDINGS_EXIT_CODE)
         elif self.semgrep_structured_errors:
-            # make a simplifying assumption that # errors = # files failed
-            # it's a quite a bit of work to simplify further because errors may or may not have path, span, etc.
-            num_errors = len(self.semgrep_structured_errors)
+            # Assumption: only the semgrep core errors pertain to files
+            semgrep_core_errors = [
+                cast(SemgrepCoreError, err)
+                for err in self.semgrep_structured_errors
+                if SemgrepError.semgrep_error_type(err) == "SemgrepCoreError"
+            ]
+            paths = set(err.path for err in semgrep_core_errors)
+            num_errors = len(paths)
```

Comment:
IMO this is worth fixing. It's pretty confusing that you can see "0 files failed" but "there were problems analyzing 1 file", as you mentioned in your comment. I vote we treat partial parse failures as full failures for now to resolve the issue, and maintain an invariant that the line you see in the normal output mode is consistent with the numbers you see under this new verbose output.

Reply:
Done